### ISSUE TITLE
MA0010 Exclude abstract classes

### ISSUE THREAD

**Issue by Paramethod** [meziantou/Meziantou.Analyzer#182]
Currently MA0010 reports 'Mark attributes with AttributeUsageAttribute' on abstract classes like this:

```
    public abstract class MyBaseAttribute : Attribute
    {

    }
```

I think the rule should exclude abstract attribute classes. What do you think?

**Comment by meziantou** [comment]
It makes sense! I think it should not report on abstract classes nor if a parent class has the attribute

````c#
// No diagnostic
abstract class MyBaseAttribute : Attribute {}
````
````c#
// No diagnostic as it has the attribute
[AttributeUsage(AttributeTargets.Property)]
class MyBaseAttribute : Attribute {}

// No diagnostic as it is set on the parent class
class MyChildAttribute : MyBaseAttribute {}
````

**Comment by Paramethod** [comment]
Maybe on parent also consider the 'inherited' value?

**Comment by meziantou** [comment]
I don't think this is needed as `AttributeUsageAttribute` is always inherited: [source](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/AttributeUsageAttribute.cs#L16).

````c#
    [AttributeUsage(AttributeTargets.Property)]
    abstract class MyCustomBaseAttribute : Attribute { }
    class MyCustomAttribute : MyCustomBaseAttribute { }

    [MyCustom] // Not valid as AttributeUsage is inherited
    class Program {}
````

**Comment by Paramethod** [comment]
Done. Thanks for the quick implementation and the project as whole.

**Comment by meziantou** [comment]
The version 1.0.544 should be available in a few minutes :)
